pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -138,26 +138,43 @@ func (r *Reconciler) reconcileIngress(ctx context.Context, dm *v1alpha1.DomainMa
 	return ingress, err
 }
 
-func (r *Reconciler) resolveRef(ctx context.Context, dm *v1alpha1.DomainMapping) (*apis.URL, error) {
+func (r *Reconciler) resolveRef(ctx context.Context, dm *v1alpha1.DomainMapping) (host string, backendSvc string, err error) {
 	resolved, err := r.resolver.URIFromKReference(ctx, &dm.Spec.Ref, dm)
 	if err != nil {
 		dm.Status.MarkReferenceNotResolved(err.Error())
-		return nil, fmt.Errorf("resolving reference: %w", err)
+		return "", "", fmt.Errorf("resolving reference: %w", err)
 	}
 
+	// Since we do not support path-based routing in domain mappings, we cannot
+	// support target references that contain a path.
 	if resolved.Path != "" {
 		dm.Status.MarkReferenceNotResolved(fmt.Sprintf("resolved URI %q contains a path", resolved))
-		return nil, fmt.Errorf("resolved URI %q contains a path", resolved)
+		return "", "", fmt.Errorf("resolved URI %q contains a path", resolved)
 	}
 
-	suffix := ".svc." + network.GetClusterDomainName()
-	if !strings.HasSuffix(resolved.Host, suffix) {
-		dm.Status.MarkReferenceNotResolved(fmt.Sprintf("resolved URI %q must end in %q", resolved, suffix))
-		return nil, fmt.Errorf("resolved URI %q must end in %q", resolved, suffix)
+	// The resolved hostname must be of the form {name}.{namespace}.svc.{suffix},
+	// which is the standard DNS address given by kubernetes to services. We will
+	// use `name` from this resolved hostname to determine the backend service
+	// name for the KIngress.
+	// TODO(julz) in the future we may support addressables that are not created
+	// from Services, in which case we would need to dynamically create the
+	// an ExternalName Service for the KIngress to use.
+	requiredSuffix := ".svc." + network.GetClusterDomainName()
```

Comment:
this looks like something that can be precomputed as global `var requiredSuffix ...`

Reply:
theoretically yeah but in practice the method has a once.Do inside it anyway. Doesn't look like we pre-compute in any of the other various places we use it, e.g. https://github.com/knative/serving/blob/ec7769beb5ac35cbc8a23d3bd18065541311d1d9/vendor/knative.dev/pkg/leaderelection/context.go#L239, https://github.com/knative/serving/blob/7398ed7b7e45bb751e6aa475a8d2f2ab15398e44/pkg/reconciler/route/config/domain.go#L104, https://github.com/knative/serving/blob/1506d1537a4047c762adc257a801c3a651f5f707/pkg/reconciler/route/domains/domains.go#L141

Reply:
One has to start somewhere :)